.github/actions/calculate-docker-image/action.yml

```diff
@@ -76,6 +76,26 @@ runs:
           echo "docker-image=${DOCKER_REGISTRY}/${REPO_NAME}/${DOCKER_IMAGE_NAME}:${DOCKER_TAG}" >> "${GITHUB_OUTPUT}"
         fi
 
+    - name: Login to registry
+      id: registry-login
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+      env:
+        DOCKER_REGISTRY: ${{ inputs.docker-registry }}
+      run: |
+        set +e
+        set -x
+
+        login() {
+          aws ecr get-login-password --region us-east-1 | docker login -u AWS --password-stdin "$1"
+        }
+
+        retry () {
+          $*  || (sleep 1 && $*) || (sleep 2 && $*)
+        }
+
+        retry login "${DOCKER_REGISTRY}"
+
     - name: Check if image should be built
       id: check-image
```

Comment:
What is the context here ? Could you please add some PR description.

There is an extra if statement in Build and push docker image  step like this, why do we want to skip it ? 
```
if: ${{ steps.calculate-image.outputs.skip != 'true' && (inputs.always-rebuild || steps.check-image.outputs.rebuild) }}
```